src/fileActions.jsx

```diff
@@ -237,3 +252,202 @@ export const RenameItemModal = ({ path, setPath, selected }) => {
+        const command = ["chmod", ...(changeAll ? ["-R"] : []), ownerAccess + groupAccess + otherAccess, "/" + path.join("/") + "/" + selected.name];
+        const renameCommand = selected.items_cnt
+            ? ["mv", "/" + path.join("/"), "/" + path.slice(0, -1).join("/") + "/" + name]
+            : ["mv", "/" + path.join("/") + "/" + selected.name, "/" + path.join("/") + "/" + name];
+
+        cockpit.spawn(command, options)
+                .then(() =>
+                    cockpit.spawn(["chown", owner + ":" + group, "/" + path.join("/") + "/" + selected.name], options)
+                            .then(() => {
+                                cockpit.spawn(renameCommand, options)
+                                        .then(() => {
+                                            if (selected.items_cnt)
+                                                setPath(path.slice(0, -1).concat(name));
+                                        });
+                            })
+                )
+                .then(Dialogs.close, err => { setErrorMessage(err.message) });
+    };
+
+    const changeOwner = (owner) => {
+        setOwner(owner);
+        const currentOwner = filteredAccounts.find(a => a.name === owner);
+        const currentGroup = filteredGroups.find(g => g.name === group);
+        if (currentGroup.gid !== currentOwner.gid && !currentGroup.userlist.includes(currentOwner.name)) {
+            setGroup(filteredGroups.find(g => g.gid === currentOwner.gid).name);
+        }
+    };
+
+    const permissions = [
+        { label: _("None"), value: "0" },
+        { label: _("Read-only"), value: "4" },
+        { label: _("Write-only"), value: "2" },
+        { label: _("Execute-only"), value: "1" },
+        { label: _("Read and write"), value: "6" },
+        { label: _("Read and execute"), value: "5" },
+        { label: _("Read, write and execute"), value: "7" },
+        { label: _("Write and execute"), value: "3" },
+    ];
+
+    return (
+        <Modal
+          position="top"
+          variant="medium"
+          title={selected.type === "file" ? _("File properties and access") : _("Directory properties and access")}
+          isOpen
+          onClose={Dialogs.close}
+          footer={
+              <>
+                  <Button variant="primary" onClick={() => editPermissions(false)}>{_("Change")}</Button>
+                  {selected.type === "directory" && <Button variant="secondary" onClick={() => editPermissions(true)}>{_("Change permissions for enclosed files")}</Button>}
+                  <Button variant="link" onClick={Dialogs.close}>{_("Cancel")}</Button>
+              </>
+          }
+        >
+            <Stack>
+                {errorMessage !== undefined &&
+                <InlineNotification
+                  type="danger"
+                  text={errorMessage}
+                  isInline
+                />}
+                <Form isHorizontal>
+                    <TextContent>
```

Comment:
Try form sections https://www.patternfly.org/components/forms/form#form-sections instead of the textcontent -> h2.